go/worker/common/p2p/p2p.go

```diff
@@ -201,6 +203,9 @@ func New(ctx context.Context, identity *identity.Identity, consensus consensus.B
 		pubsub.WithFloodPublish(true),
 		pubsub.WithPeerOutboundQueueSize(viper.GetInt(CfgP2PPeerOutboundQueueSize)),
 		pubsub.WithValidateQueueSize(viper.GetInt(CfgP2PValidateQueueSize)),
+		pubsub.WithMessageIdFn(func(pmsg *pb.Message) string {
+			return hash.NewFromBytes(pmsg.Data).String()
```

Comment:
No need to encode in hex via `String()`, just convert the hash bytes slice to a (binary) string. It should be fine here.